static/app/components/events/featureFlags/eventFeatureFlagList.tsx

```diff
@@ -68,24 +71,54 @@ export function EventFeatureFlagList({
   const {closeDrawer, isDrawerOpen, openDrawer} = useDrawer();
   const viewAllButtonRef = useRef<HTMLButtonElement>(null);
   const organization = useOrganization();
+  const eventView = useIssueDetailsEventView({group});
+  const {data: rawFlagData} = useOrganizationFlagLog({
+    organization,
+    query: {
+      start: eventView.start,
+      end: eventView.end,
+      statsPeriod: eventView.statsPeriod,
+    },
+  });
 
-  // Transform the flags array into something readable by the key-value component
-  const hydrateFlags = (flags: FeatureFlag[] | undefined): KeyValueDataContentProps[] => {
-    if (!flags) {
-      return [];
-    }
-    return flags.map(f => {
-      return {
-        item: {key: f.flag, subject: f.flag, value: f.result.toString()},
-      };
-    });
-  };
+  const {
+    suspectFlags,
+    isError: isSuspectError,
+    isPending: isSuspectPending,
+  } = useSuspectFlags({
+    organization,
+    firstSeen: group.firstSeen,
+    rawFlagData,
+    event,
+  });
 
-  // Reverse the flags to show newest at the top by default
-  const hydratedFlags = useMemo(
-    () => hydrateFlags(event.contexts?.flags?.values.reverse()),
-    [event]
-  );
+  const suspectFlagNames = useMemo(() => {
+    return isSuspectError || isSuspectPending ? [] : suspectFlags.map(f => f.flag);
+  }, [isSuspectError, isSuspectPending, suspectFlags]);
+
+  const hydratedFlags = useMemo(() => {
+    // Transform the flags array into something readable by the key-value component
+    const hydrateFlags = (
+      flags: FeatureFlag[] | undefined
+    ): KeyValueDataContentProps[] => {
+      if (!flags) {
+        return [];
+      }
+      return flags.map(f => {
+        return {
+          item: {
+            key: f.flag,
+            subject: f.flag,
+            value: f.result.toString(),
+          },
+          isSuspectFlag: suspectFlagNames.includes(f.flag),
+        };
+      });
+    };
+
+    // Reverse the flags to show newest at the top by default
+    return hydrateFlags(event.contexts?.flags?.values.reverse());
```

Comment:
something like this?
```suggestion
     const flags = event.contexts?.flags?.values.reverse() ?? [];
      return flags.map(f => {
        return {
          item: {
            key: f.flag,
            subject: f.flag,
            value: f.result.toString(),
          },
          isSuspectFlag: suspectFlagNames.includes(f.flag),
        };
      });
```

Reply:
might want like `const flags: FeatureFlag[] = [...event.contexts?.flags?.values].reverse();` so we're not reversing the original `event` object.

Reply:
i don't think `const flags: FeatureFlag[] = [...event.contexts?.flags?.values].reverse();` is working as expected, it doesn't actually reverse anything when i try 🧐

Reply:
oh, maybe try as two statements?

```
const flags: FeatureFlag[] = [...event.contexts?.flags?.values];
flags.reverse();
```

Reply:
i realized i can use `toReversed()` instead, which makes a shallow copy

Reply:
cc @scttcper for browser compat of that. should be fine imo

Reply:
its fine and/or handled by browserslist + corejs